### PR TITLE
kc protractor tests

### PR DESCRIPTION
**Pull request by kcheli**
Protractor test suite for Key Manager:
1) basic layout and functionality 
2) permissions buttons/functions on child key page 
3) key share between two accounts

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c9627fa89b9193b3cfd0d0cf2398828b80f0031e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, bug, feature**

 * Commit: 7efc3a0995c665049693b4e5fd03c9f1e10c26b6
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, bug, feature**

 * Commit: 47f177159b839ab6589e0b1830455a539c6623a8
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 374749f8045b9b44033cae9874888d59a9deef43
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: bfd0f84323f0660f54ec1a20eec08b4234e29c5f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d8629f1f0ce8f09ce680451e06ab1c5d8cd4a063
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: e7d525cb560043fc12377276aea677d13f9f59a9
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 848e21b1177ada9849d54b0af0854de742199cc2
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f0d560c1fbbb5ca99285cea11e9b5652dd7a7540
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: cd4521a7f12fa0d40a925595f25bfcbf3d25a0a4
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/Health-Nexus/KeyManager

---
This message was auto-generated by https://gitcop.com

File: protractor.conf.js [hunk 1]
```diff
@@ -1,28 +1,45 @@
...
+
 exports.config = {
-  allScriptsTimeout: 11000,
+  allScriptsTimeout: 6000000,
   specs: [
-    './e2e/**/*.e2e-spec.ts'
+    './e2e/app.basic-tests.ts',
+    './e2e/app.permissions-tests.ts',
+    './e2e/app.share-tests.ts',
+    // './e2e/**/*.e2e-spec.ts'
```
Comment (lhendre) [inline protractor.conf.js]: Delete this

File: protractor.conf.js [hunk 2]
```diff
@@ -1,28 +1,45 @@
...
   jasmineNodeOpts: {
     showColors: true,
-    defaultTimeoutInterval: 30000,
+    defaultTimeoutInterval: 6000000,
     print: function() {}
   },
+  // //Enabling blocking proxy
+  // useBlockingProxy: true,
+  // //Delaying for 3 sec before interacting with element and hightlighting it
+  // highlightDelay: 500,
```
Comment (lhendre) [inline protractor.conf.js]: If this isnt going to be added back in delete this @kcheli

File: src/app/shared/web3/drs.service.ts [hunk 3]
```diff
@@ -225,20 +224,19 @@ export class DrsService {
...
+          });
           var url='http://'+urlKey+this.unlockedAccount+'/'+signature+'/'+message_hash+'/'+parameter+'/'+key;
-          return this.http.get(url, options)
-                    .subscribe(result => {
-                      console.log('result: ', result)
-                      resolve(result);
-                    })
+          return this.http.get(url, { 
+            // headers: headers,
+            // responseType: 'blob'
```
Comment (lhendre) [inline src/app/shared/web3/drs.service.ts]: If this isnt going to be added back in delete this

File: src/app/shared/web3/drs.service.ts [hunk 4]
```diff
@@ -268,17 +266,18 @@ export class DrsService {
...
+        var headers = new HttpHeaders({ 
+          'Content-Type': 'application/octet-stream',
+        });
         var url='http://'+urlKey+'register/'+this.unlockedAccount+'/'+signature+'/'+message_hash+'/'+phuseNumber;
-        return this.http.get(url, options)
-                  .subscribe(result => {
-                    resolve(result);
-                  })
+        return this.http.get(url, {
+          // headers: headers,
```
Comment (lhendre) [inline src/app/shared/web3/drs.service.ts]: Delete this if it isnt goint to be added back in

File: src/app/shared/web3/drs.service.ts [hunk 5]
```diff
@@ -309,21 +308,22 @@ export class DrsService {
...
+          'Content-Type': 'application/octet-stream',
+        });
         var url='http://'+urlKey+'upload/'+this.unlockedAccount+'/'+signature+'/'+message_hash;
 
         const formData: FormData = new FormData();
         formData.append('fileKey', file, file.name);
         console.log("HERE", file);
-        return this.http.post(url, formData, options)
+        return this.http.post(url, formData, {
+          // headers: headers,
```
Comment (lhendre) [inline src/app/shared/web3/drs.service.ts]: See earlier comments

**Review by lhendre — COMMENTED** [review]
Fix the comments and then we should be ready to merge

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c9627fa89b9193b3cfd0d0cf2398828b80f0031e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, bug, feature**

 * Commit: 7efc3a0995c665049693b4e5fd03c9f1e10c26b6
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, bug, feature**

 * Commit: 47f177159b839ab6589e0b1830455a539c6623a8
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 374749f8045b9b44033cae9874888d59a9deef43
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: bfd0f84323f0660f54ec1a20eec08b4234e29c5f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d8629f1f0ce8f09ce680451e06ab1c5d8cd4a063
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: e7d525cb560043fc12377276aea677d13f9f59a9
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 848e21b1177ada9849d54b0af0854de742199cc2
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f0d560c1fbbb5ca99285cea11e9b5652dd7a7540
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: cd4521a7f12fa0d40a925595f25bfcbf3d25a0a4
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/Health-Nexus/KeyManager

---
This message was auto-generated by https://gitcop.com

**Comment by kcheli** [comment]
Removed the unnecessary comments.

**Comment by kcheli** [comment]
@lhendre ready to merge

**Pull request merged**